src/main/java/com/ericsson/ei/queryservice/ProcessQueryParams.java

```diff
@@ -76,13 +80,49 @@ public JSONArray filterFormParam(JSONObject criteriaObj, JSONObject optionsObj)
             String result = "{ \"$and\" : [ " + criteria + "," + options + " ] }";
             resultAggregatedObject = processAggregatedObject.processQueryAggregatedObject(result, databaseName, aggregationCollectionName);
         }
-        LOGGER.debug("resultAggregatedObject : " + resultAggregatedObject.toString());
+        if (filterKey == null || filterKey.toString().equals("{}")) {
+            LOGGER.debug("resultAggregatedObject : " + resultAggregatedObject.toString());
+        } else {
+            resultAggregatedObject = filterResult(resultAggregatedObject, filterKey);
+            LOGGER.debug("Filtered values from resultAggregatedObject : " + resultAggregatedObject.toString());
+        }
         return resultAggregatedObject;
     }
 
     /**
-     * This method takes the parameters from the REST GET request query. If the
-     * Aggregated Object matches the condition, then it is returned.
+     * This method takes array of aggregated objects and a filterKey. It returns a JSONArray where each element has a key (object Id)
+     * and a list of filtered values.
+     *
+     * @param resultAggregatedObjectArray,filterKey
+     * @return JSONArray
+     * @throws IOException
+     */
+    private JSONArray filterResult(JSONArray resultAggregatedObjectArray, JSONObject filterKey) {
+        JSONArray resultArray = new JSONArray();
+        JmesPathInterface unitUnderTest = new JmesPathInterface();
+        String searchPath = null;
+        try {
+            searchPath = filterKey.get("key").toString();
+        } catch (JSONException e1) {
+            e1.printStackTrace();
+        }
+        String processRule = "incomplete_path_filter(@, '" + searchPath + "')";
+        for (int i = 0; i < resultAggregatedObjectArray.length(); i++) {
+            try {
+                String objectId = ((JSONObject) resultAggregatedObjectArray.get(i)).get("_id").toString();
+                JsonNode filteredData = unitUnderTest.runRuleOnEvent(processRule, resultAggregatedObjectArray.get(i).toString());
+                JSONObject tempJson = new JSONObject();
+                tempJson.put(objectId, filteredData.get(searchPath).textValue());
+                resultArray.put(tempJson);
+            } catch (JSONException e) {
+                e.printStackTrace();
```

Comment:
Maybe you could create a log output here using the logger tool that is used in the rest of the code instead?

Reply:
Done